### ISSUE TITLE
Model.destroy is "undefined" in afterDestroy

### ISSUE THREAD

**Issue by fabdrol** [balderdashy/waterline#1229]
I'm seeing some strange results when unit-testing my application. According to Sails is `Model.destroy()` `undefined` when called in an afterDestroy:

```javascript
afterDestroy: function(entries, next) {
    entries = _.reduce(entries, function(list, entry) {
      sails.log.info('*** Destroyed Collection: ' + entry.title + ' (' + entry.id + ')');
      list.push(entry.id);
      return list;
    }, []);

    GroupPivot.destroy({ collection: entries })
      .then(function(destroyed) {
        return Format.destroy({ collection: entries });
      })
      .then(function(destroyed) {
        next();
      })
      .catch(function(err) {
        sails.log.error('Error destroying Formats or GroupPivots: ' + err.message);
        next(err);
      })
    ;
  }
```

Any ideas what is going on?

**Comment by fabdrol** [comment]
Note: this is during unit testing using a `sails-memory` adapter

**Comment by devinivy** [comment]
I see two `Collection.destroy()`s.  Which line is failing?  Are `GroupPivot` and `Format` waterline collections that are accessed as global variables.

**Comment by sailsbot** [comment]
Thanks for posting, @fabdrol.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
